src/Kinepolis/Trailer/YoutubeTrailerRepository.php

```diff
@@ -38,21 +44,25 @@ public function findMatchingTrailer(string $title): ?Video
             return null;
         }
 
-        $response = $this->youTubeClient->search->listSearch('id,snippet', [
-            'channelId' => $this->channelId,
-            'q' => urlencode($title),
-            'maxResults' => 1,
-        ]);
-
-        foreach ($response['items'] as $result) {
-            switch ($result['id']['kind']) {
-                case 'youtube#video':
-                    return new Video(
-                        $this->uuidGenerator->generate(),
-                        new Url('https://www.youtube.com/watch?v=' . $result['id']['videoId']),
-                        new Language('nl')
-                    );
+        try {
+            $response = $this->youTubeClient->search->listSearch('id,snippet', [
+                'channelId' => $this->channelId,
+                'q' => urlencode($title),
+                'maxResults' => 1,
+            ]);
+
+            foreach ($response['items'] as $result) {
+                switch ($result['id']['kind']) {
+                    case 'youtube#video':
+                        return new Video(
```

Comment:
Could it be useful to add an `info` log here?

Reply:
Add info about found/not found trailer